src/js/view/controls/components/menu/settings-menu.js

```diff
@@ -54,17 +69,52 @@ const SettingsMenu = (api, model, controlbar, localization) => {
 
         item.textContent = currentIndex ? '' : level.label;
     };
+    model.on('change:currentLevel', (changedModel, currentIndex) => {
+        const qualityMenu = settingsMenu.children.quality;
+        const visualQuality = model.get('visualQuality');
+        if (visualQuality && qualityMenu) {
+            changeAutoLabel(visualQuality.level, qualityMenu, currentIndex);
+        }
+        if (!qualityMenu.items[currentIndex].active) {
+            onMenuItemSelected(qualityMenu, currentIndex);
+        }
+        levelChangedManually = true;
+    }, settingsMenu);
+
+    // Visual Quality
     model.on('change:visualQuality', (changedModel, quality) => {
         const qualityMenu = settingsMenu.children.quality;
         if (quality && qualityMenu) {
             changeAutoLabel(quality.level, qualityMenu, model.get('currentLevel'));
         }
     });
-    model.on('change:currentLevel', (changedModel, currentIndex) => {
-        const qualityMenu = settingsMenu.children.quality;
-        const visualQuality = model.get('visualQuality');
-        if (visualQuality && qualityMenu) {
-            changeAutoLabel(visualQuality.level, qualityMenu, currentIndex);
+
+    // Audio Tracks
+    const setAudioTracksMenu = (audioTracks) => {
+        changeMenuItems(
+            'audioTracks', 
+            audioTracks, 
+            (index) => api.setCurrentAudioTrack(index), 
+            model.get('currentAudioTrack')
+        );
+    };
+    model.change('audioTracks', (changedModel, audioTracks) => {
+        setAudioTracksMenu(audioTracks);
+    }, settingsMenu);
+    model.on('change:currentAudioTrack', (changedModel, currentAudioTrack) => {
+        onMenuItemSelected(settingsMenu.children.audioTracks, currentAudioTrack);
+    }, settingsMenu);
+
+    // Captions
+    model.on('change:playlistItem', () => {
+        // captions.js silently clears captions when the playlist item changes. The reason it silently clear captions
+        // instead of dispatching an event is because we don't want to emit 'captionsList' if the new list is empty.
+        settingsMenu.removeMenu('captions');
```

Comment:
so if there are new captions, the captions code adds them i assume?

Reply:
Yep! Same pattern we used pre-refactor for that.